packages/apputils-extension/src/shortcuts.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { CommandRegistry } from '@lumino/commands';
+import { Selector } from '@lumino/domutils';
+import * as React from 'react';
+import { Dialog, showDialog } from '@jupyterlab/apputils';
+import { TranslationBundle } from '@jupyterlab/translation';
+
+/**
+ * The class name for each row of ContextShortcutTable
+ */
+const SHORTCUT_TABLE_ROW_CLASS = 'jp-ContextualShortcut-TableRow';
+/**
+ * The class name for the last row of ContextShortcutTable
+ */
+const SHORTCUT_TABLE_LAST_ROW_CLASS = 'jp-ContextualShortcut-TableLastRow';
+/**
+ * The class name for each item of ContextShortcutTable
+ */
+const SHORTCUT_TABLE_ITEM_CLASS = 'jp-ContextualShortcut-TableItem';
+/**
+ * The class name for each button-like symbol representing a key used in a shortcut in the ContextShortcutTable
+ */
+const SHORTCUT_KEY_CLASS = 'jp-ContextualShortcut-Key';
+/**
+ * The class name for each button-like symbol representing a key used in a shortcut in the ContextShortcutTable
+ */
+const SHORTCUT_TABLE_HEADER = 'jp-ContextualShortcut-TableHeader';
+
+export function displayShortcuts(
+  app: JupyterFrontEnd,
+  trans: TranslationBundle,
+  activeElement?: Element
+) {
```

Comment:
Two thoughts:
- narrowing public API: does `displayShortcuts` need an entire `app`? Can it just get `commands` instead?
- making API easily extensible: is it worth wrapping the arguments in an `IOptions` interface so that we can add arguments in 4.1 (if needed) without waiting for 5.0?

Reply:
Thanks @krassowski. This has been taken into account.